WorkspaceServer.Tests/PackageTests.cs

```diff
@@ -185,5 +186,17 @@ public async Task CanSupportBlazor_indicates_whether_the_package_supports_Wasm_r
             var package = await registry.Get<ICanSupportWasm>(packageName);
             package.CanSupportWasm.Should().Be(expected);
         }
+
+        [Fact]
+        public async Task If_new_files_are_added_after_one_successful_build_the_new_workspace_includes_the_file()
```

Comment:
Should probably cover the other case you added in the up to date detection (file count is same but an existing file is gone)

Reply:
I just realised that this up-to-date check will cause a serious performance regression for the Web scenario(or the non-rebuildable packages case).

Reply:
Moved the up-to-date check in the constructor of the rebuildable package which is the one used by verify and also `dotnet try`